node/external/blockAPI/baseBlock.go

```diff
@@ -278,3 +297,16 @@ func (bap *baseAPIBlockProcessor) getBlockHeaderHashAndBytesByRound(round uint64
 
 	return headerHash, blockBytes, nil
 }
+
+func extractExecutedTxHashes(mbTxHashes [][]byte, firstProcessed, lastProcessed int32) [][]byte {
+	executedTxHashes := make([][]byte, 0)
+	for txIndex, txHash := range mbTxHashes {
```

Comment:
this is a little bit of CPU processing time. A slightly optimized variant:
```GO
invalidIndexes := firstProcessed < 0 || lastProcessed >= len(mbTxHashes) || firstProcessed > lastProcessed
if invalidIndexes {
    log.Warn("extractExecutedTxHashes encountered invalid indexes", "firstProcessed", firstProcessed, 
"lastProcessed", lastProcessed, "len(mbTxHashes)", len(mbTxHashes))
    return executedTxHashes
}

for txIndex := executedTxHashes; txIndex <= lastProcessed; txIndex++{
    executedTxHashes = append(executedTxHashes, mbTxHashes[txIndex])
} 

return executedTxHashes
``` 

+add edgecases tests

Reply:
done